### PR TITLE
Feature/game detail screen

### PR DESCRIPTION
**Pull request by AShozi**
fixed the details screen scroll view

### CODE REVIEW

File: phantomgames/Search/SearchGameViewController.swift [hunk 1]
```diff
@@ -108,10 +115,21 @@ extension SearchGameViewController: UITableViewDelegate, UITableViewDataSource {
...
+    
+    public func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        guard let _ = tableView.cellForRow(at: indexPath) as? CustomTableViewCell else {
+            displayAlert(title: "Error", message: "Failed to select game. Please try again.", buttonTitle: "OK")
+            return
+        }
+        let game = viewModel.filteredGame(index: indexPath.row,
+                                          isSearchActive: searchController.isActive,
+                                          searchText: searchController.searchBar.text)
+        let gameID = game.gameID
```
Comment (ksookdhew) [inline phantomgames/Search/SearchGameViewController.swift]: This is correct but you don't necessarily need to create another local variable if its not being used again, maybe try 
performSegue(withIdentifier: Constants.SegueIdentifiers.GameDetailScreenSegue, sender: game.gameID)

**Review by ksookdhew — DISMISSED** [review]
Nice 👏

File: phantomgames/Search/SearchGameViewController.swift [hunk 2]
```diff
@@ -108,10 +115,21 @@ extension SearchGameViewController: UITableViewDelegate, UITableViewDataSource {
...
         cell.populateWith(game: newGame)
         return cell
     }
+    
+    public func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        guard let _ = tableView.cellForRow(at: indexPath) as? CustomTableViewCell else {
+            displayAlert(title: "Error", message: "Failed to select game. Please try again.", buttonTitle: "OK")
+            return
+        }
+        let game = viewModel.filteredGame(index: indexPath.row,
```
Comment (ksookdhew) [inline phantomgames/Search/SearchGameViewController.swift]: You've used this code before in the previous function, maybe look at creating a helper function and reusing it 
```
private func getSelectedGame(at index: Int) -> Game? {
    return viewModel.filteredGame(index: index,
                                  isSearchActive: searchController.isActive,
                                  searchText: searchController.searchBar.text)
}
```

File: phantomgames/Search/SearchGameViewController.swift (hunk 1 above)
Comment (AShozi) [reply phantomgames/Search/SearchGameViewController.swift]: oh ok thank you, will do

File: phantomgames/Search/SearchGameViewController.swift [hunk 3]
```diff
@@ -112,10 +119,9 @@ extension SearchGameViewController: UITableViewDelegate, UITableViewDataSource {
                 CustomTableViewCell else {
             return UITableViewCell()
         }
-        let newGame = viewModel.filteredGame(index: indexPath.row,
-                                             isSearchActive: searchController.isActive,
-                                             searchText: searchController.searchBar.text)
-        cell.populateWith(game: newGame)
```
Comment (ksookdhew) [inline phantomgames/Search/SearchGameViewController.swift:112]: Nice 👏

**Review by ksookdhew — APPROVED** [review]

**Review by MarcusNkadimeng — APPROVED** [review]
Nice!

**Pull request merged**